src/QMCWaveFunctions/WaveFunctionComponent.h

```diff
@@ -347,10 +362,7 @@ class WaveFunctionComponent : public QMCTraits
                                     int iat,
                                     std::vector<PsiValueType>& ratios,
                                     std::vector<GradType>& grad_new,
-                                    std::vector<ComplexType>& spingrad_new) const
-  {
-    mw_ratioGrad(wfc_list, p_list, iat, ratios, grad_new);
-  }
+                                    std::vector<ComplexType>& spingrad_new) const;
```

Comment:
Is this function still been called from outside? otherwise, let us move it under private.

Reply:
right now none of the mw_ratioGradWithSpin or mw_evalGradWithSpin has actually been implemented for DiracDet or MultiDiracDet. That is for a future PR. 

I think if this goes to private, it would make sense to move the "non-spin" version as private as well and only keep the templated API as public.

Reply:
Yes. If no existing calls from outside, move both to private and only expose the template one.

Reply:
the WithSpin APIs are now private.